README.md

```diff
@@ -139,52 +134,117 @@ Previous simulation platforms that have operated on similar datasets typically p
 
-    **Systems with CUDA**
 
-    Add the `--with-cuda` flag, i.e. `python setup.py install --with-cuda`, to build CUDA features
+1. Clone this github repository. 
+   ```bash
+   # Clone and checkout the latest stable release
+   git clone --branch v0.1.2 https://github.com/facebookresearch/habitat-sim.git
+   cd habitat-sim
+   ```
+   
+   List of stable releases is [available here](https://github.com/facebookresearch/habitat-sim/releases). Master branch contains 'bleeding edge' code and under active development. 
+
+1. Install Dependencies 
+  
+    Common
+   ```bash
+   # We require python>=3.6 and cmake>=3.10
+   conda create -n habitat python=3.6 cmake=3.14.0
+   conda activate habitat
+   pip install -r requirements.txt
+   ```
+
+    Linux (Tested with Ubuntu 18.04 with gcc 7.4.0)
+   ```bash
+   sudo apt-get update || true
+   # These are fairly ubiquitous packages and your system likely has them already, 
+   # but if not, let's get the essentials for EGL support:
+   sudo apt-get install -y --no-install-recommends \
+        libjpeg-dev libglm-dev libgl1-mesa-glx libegl1-mesa-dev mesa-utils xorg-dev freeglut3-dev
+   ```
+
+   See this [configuration for a full list of dependencies](https://github.com/facebookresearch/habitat-sim/blob/master/.circleci/config.yml#L64) that our CI installs on a clean Ubuntu VM. If you run into build errors later, this is a good place to check if all dependencies are installed. 
+
+1. Build Habitat-Sim 
+
+    Default build (for machines with a display attached)
+   ```bash
+   # Assuming we're still within habitat conda environment
+   python setup.py install
+   ```
+
+    For headless systems (i.e. without an attached display, e.g. in a cluster) and multiple GPU systems
+   ```bash
+   python setup.py install --headless
+   ```
+
+    For systems with CUDA (to build CUDA features)
+   ```bash
+   python setup.py install --with-cuda
+   ```
+
+   (Under development) With physics simulation via [Bullet Physics SDK](https://github.com/bulletphysics/bullet3/): 
+   First, install [Bullet Physics](https://github.com/bulletphysics/bullet3/). Next use
+   ```bash
+   python setup.py install --bullet    # build habitat with bullet physics
+   ```
+
+   Note1: some Linux distributions might require an additional `--user` flag to deal with permission issues. 
+   
+   Note2: for active development in Habitat, you might find `./build.sh` instead of `python setup.py install` more useful. 
+
+
+1. [Optional; Needed only with `build.sh`] For use with [Habitat-API](https://github.com/facebookresearch/habitat-api) and your own python code, add habitat-sim to your `PYTHONPATH`. For example modify your `.bashrc` (or `.bash_profile` in Mac OS X) file by adding the line:
```

Comment:
```suggestion
1. [Only if using `build.sh`] For use with [Habitat-API](https://github.com/facebookresearch/habitat-api) and your own python code, add habitat-sim to your `PYTHONPATH`. For example modify your `.bashrc` (or `.bash_profile` in Mac OS X) file by adding the line:
```

Doing that can actually cause things to break when using `setup.py install` as sometimes the python code and bindings will get out of sync without you realizing and/or the bindings won't get picked up at all.